src/Groups/matrices/matrix_manipulation.jl

```diff
@@ -0,0 +1,276 @@
+import AbstractAlgebra: FieldElem, map, Ring
+import Hecke: evaluate, multiplicative_jordan_decomposition, PolyElem, _rational_canonical_form_setup, refine_for_jordan
+
+export
+    block_matrix,
+    complement,
+    conjugate_transpose,
+    diagonal_join,
+    insert_block,
+    insert_block!,
+    isconjugate_gl,
+    ishermitian_matrix,
+    isskewsymmetric_matrix,
+    partitions,
+    permutation_matrix,
+    submatrix
+
+
+
+
+function partitions(n::Int, m::Int)         # partitions where the biggest block is at most m
+   if n==0 return [Int[]] end
+   if m==1 return [[1 for i in 1:n]] end
+   L = []
+   for i in 0:div(n,m)
+      t = [m for i in 1:i]
+      for k in partitions(n-m*i, m-1)
+         push!(L, vcat(t,k))
+      end
+   end
+
+   return L
+end
```

Comment:
Just a warning:
Uli Thiel's [JuLie package](https://github.com/ulthiel/JuLie.jl) als defines a function `partitions`, which returns the partitions in a different order.

Reply:
If it is accessible from Oscar, we can substitute my `partitions` with the JuLie package's partitions.

Reply:
The plan is to use JuLie together with Oscar, I do not know whether it will become a part of Oscar or a dependency of Oscar. The main point for me is that it will be confusing for users to have different meanings of `partitions` in different contexts. Now we still have the chance to avoid this confusion.